client/my-sites/media-library/upload-controls.jsx

```diff
@@ -0,0 +1,186 @@
+	 * but is supported in Internet Explorer and Chrome browsers. Further input
+	 * validation will occur when attempting to upload the file.
+	 *
+	 * @return {string} Supported file extensions, as comma-separated string
+	 */
+	getInputAccept = () => {
+		if ( ! MediaUtils.isSiteAllowedFileTypesToBeTrusted( this.props.site ) ) {
+			return null;
+		}
+		const allowedFileTypesForSite = MediaUtils.getAllowedFileTypesForSite( this.props.site );
+
+		return uniq( allowedFileTypesForSite.concat( VideoPressFileTypes ) ).map( ( type ) => `.${ type }` ).join();
+	};
+
+	renderUploadViaUrl() {
+		const { site, translate, onAddMedia } = this.props;
+
+		if ( this.state.addingViaUrl ) {
+			return (
+				<CompactCard className="media-library__upload-controls-option">
+					<MediaLibraryUploadUrl
+						site={ site }
+						onAddMedia={ onAddMedia }
+						onClose={ this.onAddUrlClose } />
+				</CompactCard>
+			);
+		}
+
+		return (
+			<CompactCard className="media-library__upload-controls-option" onClick={ this.onAddUrlClick }>
+				<Gridicon icon="domains" size={ 18 } />
+				<span className="media-library__upload-controls-option-name">{ translate( 'URL' ) }</span>
+				<Gridicon icon="chevron-right" />
+			</CompactCard>
+		);
+	}
+
+	renderFileItem() {
+		const { translate } = this.props;
+
+		return (
+			<form ref="itemForm">
+				<CompactCard className="media-library__upload-controls-option">
+					<Gridicon icon="computer" size={ 18 } />
+					<span className="media-library__upload-controls-option-name">{ translate( 'Computer' ) }</span>
+					<Gridicon icon="chevron-right" />
+					<input
+						className="media-library__upload-controls-file-input"
+						type="file"
+						accept={ this.getInputAccept() }
+						multiple
+						onChange={ this.uploadFiles }
+						onClick={ this.onUploadClick } />
+				</CompactCard>
+			</form>
+		);
+	}
+
+	renderDropzone() {
+		const { translate } = this.props;
+
+		return (
+			<div className="media-library__upload-controls-drop-zone-container">
+				<Gridicon icon="cloud-upload" size={ 48 } />
+				<h3>{ translate( 'Drag and drop' ) }</h3>
+				<form ref="dropzoneForm">
+				<p>{ translate( 'Your files anywhere to upload, or {{span}}{{input/}}browse{{/span}}', {
+					components: {
+						span: <span className="media-library__upload-controls-file-anchor" />,
+						input: <input
+								className="media-library__upload-controls-file-input"
+								type="file"
+								accept={ this.getInputAccept() }
+								multiple
+								onChange={ this.uploadFiles }
+								onClick={ this.onUploadClick } />
+					} } ) }</p>
+				</form>
+			</div>
+		);
+	}
+
+	render() {
+		const { translate, site } = this.props;
+
+		if ( ! userCan( 'upload_files', site ) ) {
+			return null;
+		}
+
+		return (
+			<div className="media-library__upload-controls-container">
+				<div>
+					<SectionHeader
+						className="media-library__upload-controls-header"
+						label={ translate( 'Upload from' ) } />
```

Comment:
Hi! I've found a possible matching string that has already been translated **104** times:
`translate( 'Upload' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).